clusterloader2/pkg/measurement/common/kcm_metrics.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"k8s.io/klog/v2"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/provider"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	kcmWorkQueueDepthMetricName    = "KCMWorkQueueDepthMetrics"
+	kcmWorkQueueDepthMetricVersion = "v1"
+	hpaResource                    = "horizontalpodautoscaler"
+)
+
+func init() {
+	if err := measurement.Register(kcmWorkQueueDepthMetricName, createKCMWorkQueueDepthMeasurement); err != nil {
```

Comment:
Given this file is named as `kcm_metrics.go`, can we make it generic and flexible to measure other KCM metrics in future if we want ? Instead of pinning it down to `workqueuedepth` ? 

`
if err := measurement.Register(kcmMetricsName, createKCMMetricsMeasurement); err != nil { 
klog.Fatalf("Cannot register %s: %v", kcmMetricsName, err)
	}
}
`
wdyt ?